drivers/spi/spi_ll_stm32.c

```diff
@@ -88,7 +94,12 @@ static const struct mem_region nocache_mem_regions[] = {
 /* dummy value used for transferring NOP when tx buf is null
  * and use as dummy sink for when rx buf is null
  */
-uint32_t dummy_rx_tx_buffer;
+#ifdef CONFIG_NOCACHE_MEMORY
+static __aligned(32) uint32_t dummy_rx_tx_buffer
```

Comment:
Running the 16 bit frames tests has revealed a bug in the SPI driver: the dummy_rx_tx_buffer is not in a nocache region. When DMA is enabled, this means that his variable's value will not necessarily be zero. This is a problem as the dummy buffer should always be 0, so SPI sends zeroes whenever required (there are a number of situations in which this is required, see spi.h interface. In summary, when rx is finished or NULL but tx is not, and vice-versa).

The solution to this problem is setting the variable in a nocache region, but this can only be done when CONFIG_NOCACHE_MEMORY. When it is not, the nocache regions are defined in devicetree, but are unknown to this driver. Furthermore, there could be several nocache regions declared there, and it's not clear which one should be used, even if it could be accessed.

To solve this, if CONFIG_NOCACHE_MEMORY is not enabled, don't place the variable in a nocache area. Instead, keep cache coherency manually.

Reply:
A summary of this comment would be valuable in code.

Reply:
I added some comments. Hopefully they are helpful.